src/main/java/com/terraforged/mod/server/command/TerraCommand.java

```diff
@@ -242,32 +265,44 @@ private static int findTerrainAndBiome(CommandContext<CommandSource> context) th
         Search biomeSearch = new BiomeSearchTask(pos, biome, getChunkGenerator(context), getBiomeProvider(context));
         Search terrainSearch = new TerrainSearchTask(pos, target, getChunkGenerator(context), generator);
         Search search = new BothSearchTask(pos, biomeSearch, terrainSearch);
-        doSearch(server, playerID, search);
-        context.getSource().sendFeedback(new StringTextComponent("Searching..."), false);
+        int identifier = doSearch(server, playerID, search);
+        context.getSource().sendFeedback(createIdentifierMessage(identifier)
+                                         .appendSibling(new StringTextComponent(" Searching for "))
+                                         .appendSibling(createBiomeDisplayMessage(biome))
+                                         .appendSibling(new StringTextComponent(" and "))
+                                         .appendSibling(createTerrainMessage(target))
+                                         .appendSibling(new StringTextComponent("..."))
+                                         , false);
 
         return Command.SINGLE_SUCCESS;
     }
 
-    private static void doSearch(MinecraftServer server, UUID userId, Supplier<BlockPos> supplier) {
+    private static int doSearch(MinecraftServer server, UUID userId, Supplier<BlockPos> supplier) {
+        searchId.putIfAbsent(userId, 0);
+        int identifier = searchId.get(userId);
```

Comment:
This can be `int identifier = getOrDetault(userId, 0)` since there's a put happening later